lib/THC/THCTensorMath.cu

```diff
@@ -759,32 +627,33 @@ void THCudaTensor_cumprod(THCState *state, THCudaTensor *self, THCudaTensor *src
   return THCudaTensor_scanDim(state, self, src, dimension, 1.0f, thrust::multiplies<float>());
 }
 
-/* A set of reduction kernels that take in binary ops on thrust pairs (of value, index).
+/* a set of reduction kernels that take in Binary ops on thrust pairs (of value, index)
    These are useful when you not only have to do a reduction, but you might have
-   to preserve the location of contention (for example min/max operations).
-   The structure of the kernels follows the structure of the reduction kernels.
-*/
+   to preserve the location of contention (for example min/max operations)
+ */
 template<class BinaryFunction>
 __global__ void THCudaTensor_kernel_transformReduceOuterDimIndex(float *tgt1, float *tgt2,
                                                              float *src_,
-                                                             unsigned num_orows,
-                                                             unsigned num_irows,
-                                                             unsigned row_size,
+                                                             dim4 src_stride,
```

Comment:
yes, @soumith merged from an earlier point since our internal tree wasn't quite updated with latest. See my "merge issues here, since we maybe didn't have the latest? i didn't change this code." comments up above. He'll fix this.